src/components/EntitlementLoader.js

```diff
@@ -0,0 +1,226 @@
+import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { okapi } from 'stripes-config';
+import { useStripes } from '../StripesContext';
+import { ModulesContext, useModules } from '../ModulesContext';
+import loadRemoteComponent from '../loadRemoteComponent';
+
+/**
+ * preloadModules
+ * Loads each module code and sets up its getModule function.
+ * Map the list of applications to a hash keyed by acts-as type (app, plugin,
+ * settings, handler) where the value of each is an array of corresponding
+ * applications.
+ *
+ * @param {array} remotes
+ * @returns {app: [], plugin: [], settings: [], handler: []}
+ */
+
+const preloadModules = async (stripes, remotes) => {
+  const modules = { app: [], plugin: [], settings: [], handler: [] };
+
+  try {
+    const loaderArray = [];
+    remotes.forEach(async remote => {
+      const { name, url } = remote;
+      loaderArray.push(loadRemoteComponent(url, name)
+        .then((module) => {
+          remote.getModule = () => module.default;
+        })
+        .catch((e) => { throw new Error(`Error loading code for remote module: ${name}: ${e}`); }));
+    });
+
+    await Promise.all(loaderArray);
+
+    // once the all the code for the modules are loaded, populate the `modules` structure based on `actsAs` keys.
+    remotes.forEach((remote) => {
+      const { actsAs } = remote;
+      actsAs.forEach(type => modules[type].push({ ...remote }));
+    });
+  } catch (e) {
+    stripes.logger.log('core', `Error preloading modules from entitlement response: ${e}`);
+  }
+
+  return modules;
+};
+
+/**
+ * loadTranslations
+ * return a promise that fetches translations for the given module,
+ * dispatches setLocale, and then returns the translations.
+ *
+ * @param {object} stripes
+ * @param {object} module info read from the registry
+ *
+ * @returns {Promise}
+ */
+const loadTranslations = (stripes, module) => {
+  // construct a fully-qualified URL to load.
+  //
+  // locale strings include a name plus optional region and numbering system.
+  // we only care about the name and region. this stripes the numberin system
```

Comment:
this is probably my comment that got cut-and-pasted, complete with typos 🤦 . Should be "this strips the numbering system..."